pkg/cmd/codespace/ssh.go

```diff
@@ -147,6 +155,129 @@ func (a *App) SSH(ctx context.Context, sshArgs []string, opts sshOptions) (err e
 	}
 }
 
+func (a *App) ListOpensshConfig(ctx context.Context, opts configOptions) error {
+	// Ensure all child tasks (e.g. port forwarding) terminate before return.
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	var err error
+	var codespaces []*api.Codespace
+	if opts.codespace == "" {
+		a.StartProgressIndicatorWithLabel("Fetching codespaces")
+		codespaces, err = a.apiClient.ListCodespaces(ctx, -1)
+		a.StopProgressIndicator()
+	} else {
+		var codespace *api.Codespace
+		codespace, err = getOrChooseCodespace(ctx, a.apiClient, opts.codespace)
+		codespaces = []*api.Codespace{codespace}
+	}
+	if err != nil {
+		return fmt.Errorf("error getting codespace info: %w", err)
+	}
+
+	t, err := template.New("ssh_config").Parse(`Host cs.{{.Name}}.{{.EscapedRef}}
+	User {{.SshUser}}
+	ProxyCommand {{.GHExec}} cs ssh -c {{.Name}} --stdio
+	UserKnownHostsFile=/dev/null
+	StrictHostKeyChecking no
+	LogLevel quiet
+	ControlMaster auto
+
+`)
+	if err != nil {
+		return fmt.Errorf("error formatting template: %w", err)
+	}
+
+	ghexec, err := os.Executable()
+	if err != nil {
+		return err
+	}
+
+	sshUsers := map[string]string{}
+
+	for _, cs := range codespaces {
+
+		if cs.State != "Available" {
+			fmt.Fprintf(os.Stderr, "skipping unavailable codespace %s: %s\n", cs.Name, cs.State)
+			continue
+		}
+
+		var sshUser string
+		var ok bool
+
+		if sshUser, ok = sshUsers[cs.Repository.FullName]; !ok {
+			session, err := openSshSession(ctx, a, cs.Name, nil)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "error connecting to codespace: %v\n", err)
```

Comment:
Shouldn't this cause a nonzero exit?

Reply:
I added an [explanation](https://github.com/cli/cli/pull/4915/commits/369ea534b67add703502849909db3c7f2767e277) - I didn't want a problem with one codespace to prevent connecting to any of them.

You're right though, we should at least emit a nonzero exit code in this case.

Reply:
Here's an example. Note that we end up with 2 host configs while skipping the third, and that the generated configs have different `User` directives.
```
[git::znull/stdio](0)➤ gh cs list
NAME                               REPOSITORY       BRANCH   STATE      CREATED AT
znull-github-codeload-r7wrxg72447  github/codeload  master   Shutdown   2h
znull-github-babeld-4q6gr7qfqjjp   github/babeld    master*  Available  2h
znull-github-github-77wrx972xg6    github/github    master   Available  20m

[git::znull/stdio](0)➤ gh cs ssh config > ~/.config/ssh.d/codespaces
skipping unavailable codespace znull-github-codeload-r7wrxg72447: Shutdown

[git::znull/stdio](0)➤ cat ~/.config/ssh.d/codespaces
Host cs.znull-github-babeld-4q6gr7qfqjjp.master
        User codespace
        ProxyCommand /home/znull/src/cli/bin/gh cs ssh -c znull-github-babeld-4q6gr7qfqjjp --stdio
        UserKnownHostsFile=/dev/null
        StrictHostKeyChecking no
        LogLevel quiet
        ControlMaster auto

Host cs.znull-github-github-77wrx972xg6.master
        User root
        ProxyCommand /home/znull/src/cli/bin/gh cs ssh -c znull-github-github-77wrx972xg6 --stdio
        UserKnownHostsFile=/dev/null
        StrictHostKeyChecking no
        LogLevel quiet
        ControlMaster auto
```

Reply:
Current behavior of warnings being printed to stderr and the command continuing to print config for other codespaces is fine. If there were any warnings, however, you might choose to exit with nonzero status. You can return `cmd.SilentError` in the end if you want that behavior.

Reply:
> I didn't want a problem with one codespace to prevent connecting to any of them

Agreed: we should print partial results, but a command should always return nonzero if it didn't complete its whole task.

(BTW, don't forget that the cli/cli repo is public, so it's best to avoid using names of private repositories (no matter how obscure their names) in the examples.)